src/models/lstm/LSTM.cpp

```diff
@@ -0,0 +1,240 @@
+     * `manage_state_before_next_time_step_run`.
+     *
+     * @param dt the time step size in seconds
+     * @param DLWRF_surface_W_per_meters_squared
+     * @param PRES_surface_Pa
+     * @param SPFH_2maboveground_kg_per_kg
+     * @param precip_meters_per_second
+     * @param DSWRF_surface_W_per_meters_squared
+     * @param TMP_2maboveground_K
+     * @param UGRD_10maboveground_meters_per_second
+     * @param VGRD_10maboveground_meters_per_second  
+     * @return
+     */
+    int lstm_model::run(double dt, double DLWRF_surface_W_per_meters_squared,
+                        double PRES_surface_Pa, double SPFH_2maboveground_kg_per_kg,
+                        double precip_meters_per_second, double DSWRF_surface_W_per_meters_squared,
+                        double TMP_2maboveground_K, double UGRD_10maboveground_meters_per_second,
+                        double VGRD_10maboveground_meters_per_second) {
+        // Do resetting/housekeeping for new calculations and new state values
+
+        manage_state_before_next_time_step_run();
+
+        std::vector<torch::jit::IValue> inputs;
+        torch::Tensor forcing = torch::zeros({1, 11});
+
+        forcing[0][0] = lstm_model::normalize("Precip_rate", precip_meters_per_second);
+        forcing[0][1] = lstm_model::normalize("SPFH_2maboveground_kg_per_kg", SPFH_2maboveground_kg_per_kg);
+        forcing[0][2] = lstm_model::normalize("TMP_2maboveground_K", TMP_2maboveground_K);
+        forcing[0][3] = lstm_model::normalize("DLWRF_surface_W_per_meters_squared", DLWRF_surface_W_per_meters_squared);
+        forcing[0][4] = lstm_model::normalize("DSWRF_surface_W_per_meters_squared", DSWRF_surface_W_per_meters_squared);
+        forcing[0][5] = lstm_model::normalize("PRES_surface_Pa", PRES_surface_Pa);
+        forcing[0][6] = lstm_model::normalize("UGRD_10maboveground_meters_per_second", UGRD_10maboveground_meters_per_second);
+        forcing[0][7] = lstm_model::normalize("VGRD_10maboveground_meters_per_second", VGRD_10maboveground_meters_per_second);
+        forcing[0][8] = lstm_model::normalize("Area_Square_km", model_params.area);
+        forcing[0][9] = lstm_model::normalize("Latitude", model_params.latitude);
+        forcing[0][10] = lstm_model::normalize("Longitude", model_params.longitude);
+
+        // Create the model input for one time step
+      	inputs.push_back(forcing.to(device));
+        inputs.push_back(previous_state->h_t);
+        inputs.push_back(previous_state->c_t);
+      	// Run the model
+        auto output = model.forward(inputs).toTuple()->elements();
+      	//Get the outputs
+        double out_flow = lstm_model::denormalize( "obs", output[0].toTensor().item<double>() );
+        out_flow = out_flow * 0.028316847; //convert cfs to cms
+        fluxes = std::make_shared<lstm_fluxes>( lstm_fluxes( out_flow ) ) ;
+
+        current_state = std::make_shared<lstm_state>( lstm_state(output[1].toTensor(), output[2].toTensor()) );
+
+        return 0;
```

Comment:
Need to document the  return meaning of this function (i.e. 0 mean run successful, something else  is error)

Reply:
Add this to the doc string for the function under `@return`